src/test/java/org/gaul/s3proxy/CrossOriginResourceSharingResponseTest.java

```diff
@@ -0,0 +1,305 @@
+import org.junit.Test;
+
+public final class CrossOriginResourceSharingResponseTest {
+    static {
+        System.setProperty(
+                SDKGlobalConfiguration.DISABLE_CERT_CHECKING_SYSTEM_PROPERTY,
+                "true");
+        AwsSdkTest.disableSslVerification();
+    }
+
+    private URI s3Endpoint;
+    private EndpointConfiguration s3EndpointConfig;
+    private S3Proxy s3Proxy;
+    private BlobStoreContext context;
+    private String blobStoreType;
+    private String containerName;
+    private AWSCredentials awsCreds;
+    private AmazonS3 s3Client;
+    private String servicePath;
+    private CloseableHttpClient httpClient;
+    private URI presignedGET;
+
+    @Before
+    public void setUp() throws Exception {
+        TestUtils.S3ProxyLaunchInfo info = TestUtils.startS3Proxy(
+                "s3proxy-cors.conf");
+        awsCreds = new BasicAWSCredentials(info.getS3Identity(),
+                info.getS3Credential());
+        context = info.getBlobStore().getContext();
+        s3Proxy = info.getS3Proxy();
+        s3Endpoint = info.getSecureEndpoint();
+        servicePath = info.getServicePath();
+        s3EndpointConfig = new EndpointConfiguration(
+                s3Endpoint.toString() + servicePath, "us-east-1");
+        s3Client = AmazonS3ClientBuilder.standard()
+                .withCredentials(new AWSStaticCredentialsProvider(awsCreds))
+                .withEndpointConfiguration(s3EndpointConfig)
+                .build();
+        httpClient = getHttpClient();
+
+        containerName = createRandomContainerName();
+        info.getBlobStore().createContainerInLocation(null, containerName);
+
+        String blobName = "test";
+        ByteSource payload = ByteSource.wrap("blob-content".getBytes(
+                StandardCharsets.UTF_8));
+        Blob blob = info.getBlobStore().blobBuilder(blobName)
+                .payload(payload).contentLength(payload.size()).build();
+        info.getBlobStore().putBlob(containerName, blob);
+
+        Date expiration = new Date(System.currentTimeMillis() +
+                TimeUnit.HOURS.toMillis(1));
+        presignedGET = s3Client.generatePresignedUrl(containerName, blobName,
+                expiration, HttpMethod.GET).toURI();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        if (s3Proxy != null) {
+            s3Proxy.stop();
+        }
+        if (context != null) {
+            context.getBlobStore().deleteContainer(containerName);
+            context.close();
+        }
+        if (httpClient != null) {
+            httpClient.close();
+        }
+    }
+
+    @Test
+    public void testCorsPreflightNegative() throws Exception {
+        // No CORS headers
+        HttpOptions request = new HttpOptions(presignedGET);
+        CloseableHttpResponse response = httpClient.execute(request);
+        assertThat(response.getStatusLine().getStatusCode())
+                .isEqualTo(HttpStatus.SC_FORBIDDEN);
```

Comment:
When running s3-tests via `./virtualenv/bin/nosetests -a 'cors'`, I see four errors of the form:

```
Traceback (most recent call last):
  File "/home/gaul/work/s3proxy/s3-tests/virtualenv/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/gaul/work/s3proxy/s3-tests/s3tests/functional/test_s3.py", line 6125, in test_cors_header_option
    eq(e.status, 404)
AssertionError: 403 != 404
```

I lack background in CORS but is this the correct error code?

Reply:
please see my longer comment here https://github.com/gaul/s3proxy/pull/287#issuecomment-449501179